src/client/v2/indexer/searchForApplicationBoxes.ts

```diff
@@ -0,0 +1,75 @@
+import JSONRequest from '../jsonrequest';
+import HTTPClient from '../../client';
+import IntDecoding from '../../../types/intDecoding';
+
+export default class SearchForApplicationBoxes extends JSONRequest {
+  /**
+   * Returns information about indexed application boxes.
+   *
+   * #### Example
+   * ```typescript
+   * const boxesResult = await indexerClient.SearchForApplicationBoxes(1234).do();
+   * ```
+   *
+   * [Response data schema details](https://developer.algorand.org/docs/rest-apis/indexer/#get-v2applicationsapplication-idboxes)
+   * @oaram index - application index.
+   * @category GET
+   */
+  constructor(c: HTTPClient, intDecoding: IntDecoding, private index: number) {
+    super(c, intDecoding);
+    this.index = index;
+  }
+
+  /**
+   * @returns `/v2/applications/${index}/boxes`
+   */
+  path() {
+    return `/v2/applications/${this.index}/boxes`;
+  }
+
+  /**
+   * Specify the next page of results.
+   *
+   * #### Example
+   * ```typescript
+   * const maxResults = 20;
+   *
+   * const boxesPage1 = await indexerClient
+   *        .SearchForApplicationBoxes(1234)
+   *        .limit(maxResults)
+   *        .do();
+   *
+   * const boxesPage2 = await indexerClient
+   *        .SearchForApplicationBoxes(1234)
+   *        .limit(maxResults)
+   *        .nextToken(boxesPage1["next-token"])
+   *        .do();
+   * ```
+   * @param nextToken - provided by the previous results.
+   * @category query
+   */
+  nextToken(next: string) {
+    this.query.next = next;
```

Comment:
```suggestion
    this.query.next = "b64:" + next;
```
I'm guessing that this addresses the same issue which [I brought up in the Python SDK](https://github.com/algorand/py-algorand-sdk/pull/366/files#r948447135).

Reply:
I verified that this works even in the empty `next` case: https://github.com/algorand/indexer/blob/tzaffi/box-search/api/test_resources/boxes.json#L886-L887

Reply:
From the decision we made in https://github.com/algorand/algorand-sdk-testing/pull/219, we keep the `nextToken` method unchanged.